tests/unit/store/unload-test.js

```diff
@@ -84,9 +84,10 @@ test("unload a record", function(assert) {
 
       run(function() {
         store.unloadRecord(record);
+        assert.equal(get(record, 'hasDirtyAttributes'), false, "record is not dirty");
       });
 
-      assert.equal(get(record, 'hasDirtyAttributes'), false, "record is not dirty");
```

Comment:
The reason this assertion is failing is due to the implementation of `hasDirtyAttributes` in [`DS.Model`](https://git.io/vrGlF). It looks like a change to the `currentState.isDirty` dependency of the computed property is triggered when the record is unloaded.

@emberjs/ember-data-contributors are there any objections in changing the implementation so it re-uses the [`retrieveFromCurrentState`](https://git.io/vrGlp) function, like we already do for the other flags like `isDirty`?